### PR TITLE
refactor(ext/ffi): split into multiple parts

### PR DESCRIPTION
**Pull request by littledivy**
- [x] `dlfcn.rs` - `dlopen()`-related code.
- [x] `turbocall.rs` - Call trampoline JIT compiler. 
- [x] `repr.rs` - Pointer representation. Home of the UnsafePointerView ops.
- [x] `symbol.rs` - Function symbol related code.
- [x] `callback.rs` - Home of `Deno.UnsafeCallback` ops.
- [x] `ir.rs` - Intermediate representation for values. Home of the `NativeValue` type.
- [x] `call.rs` - Generic call ops. Home to everything related to calling FFI symbols.
- [x] `static.rs` - static symbol support

I find easier to work with this setup, I eventually want to expand TurboCall to unroll type conversion loop in generic calls, generate code for individual symbols (lazy function pointers), etc.

Also, yes... for the lack of a catchy name, I'm renaming `fast_call.rs` to turbocall. Suggestions are welcome.

### CODE REVIEW

**Comment by aapoalas** [comment]
How about `dynocall` for Demo, and Dyno power generation and dynamism? :D

**Comment by lino-levan** [comment]
Personally more of a fan of `fast_call`. `turbocall` reminds me too much of vercel branding and `dynocall` reminds me too much of heroku branding.

**Comment by littledivy** [comment]
`turbocall` made up from: V8's **turbo**fan **call** trampoline. IMO It's not something to brand really, just easy way to refer in developement. `fast_call` means a lot of things in the codebase, when I refer to FFI fast calls I say _turbocall_

File: ext/ffi/call.rs [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+  ForeignFunction,
+};
+use deno_core::{
+  anyhow::anyhow,
+  error::{type_error, AnyError},
+  op,
+  serde_json::Value,
+  serde_v8, v8, ResourceId,
+};
+use libffi::middle::Arg;
```
Comment (bartlomieju) [inline ext/ffi/call.rs:17]: Rewrite to single import per line

File: ext/ffi/callback.rs [hunk 2]
```diff
@@ -0,0 +1,556 @@
...
+  check_unstable, FfiPermissions, FfiState, ForeignFunction,
+  PendingFfiAsyncWork, LOCAL_ISOLATE_POINTER, MAX_SAFE_INTEGER,
+  MIN_SAFE_INTEGER,
+};
+use deno_core::futures::channel::mpsc;
+use deno_core::CancelFuture;
+use deno_core::{
+  error::AnyError, op, serde_v8, v8, CancelHandle, OpState, Resource,
+  ResourceId,
+};
```
Comment (bartlomieju) [inline ext/ffi/callback.rs:14]: Ditto

File: ext/ffi/dlfcn.rs [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+  check_unstable,
+  symbol::{NativeType, Symbol},
+  FfiPermissions,
+};
+use deno_core::{
+  error::{generic_error, AnyError},
+  op, serde_v8, v8, Resource, ResourceId,
+};
+use dlopen::raw::Library;
+use serde::Deserialize;
```
Comment (bartlomieju) [inline ext/ffi/dlfcn.rs:14]: Ditto

File: ext/ffi/ir.rs [hunk 4]
```diff
@@ -0,0 +1,525 @@
...
+
+use crate::symbol::NativeType;
+use crate::{MAX_SAFE_INTEGER, MIN_SAFE_INTEGER};
+use deno_core::{
+  error::{type_error, AnyError},
+  serde_json::Value,
+  serde_v8, v8,
+};
+use libffi::middle::Arg;
+use std::{ffi::c_void, ptr};
```
Comment (bartlomieju) [inline ext/ffi/ir.rs:11]: Please update to single import per line in all files

**Review by bartlomieju — APPROVED** [review]
LGTM, if the comments are addressed. This is a very nice cleanup, I think it's very beneficial to have these split as FFI is probably one of more complex parts of the system

**Pull request merged**